app.rb

```diff
@@ -0,0 +1,176 @@
+require_relative 'person'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'teacher'
+require_relative 'student'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def book_list
+    puts 'List of all books'
+    @books.each do |book|
+      puts "Title: \"#{book.title}\" , Author: #{book.author}"
+    end
+  end
+
+  def people_list
+    puts 'List of all people'
+    @people.each do |person|
+      puts person
+    end
+  end
+
+  def create_student(age, name, parent_permission)
+    student = Student.new(age, name, parent_permission)
+    @people << student
+  end
+
+  def create_teacher(age, name, specialization)
+    teacher = Teacher.new(age, name, specialization)
+    @people << teacher
+  end
+
+  def create_book(title, author)
+    book = Book.new(title, author)
+    @books << book
+  end
+
+  def create_rental(date, book, person)
+    rental = Rental.new(date, book, person)
+    @rentals << rental
+  end
```

Comment:
- [ ] Kindly make sure that a user can create a rental, There is an error when I try to create a rental, see the error below ⬇️  

  
![Screenshot_167](https://github.com/Innocent259/School-library/assets/69638013/39c23584-ef0d-494b-8bcb-f4d0a4fa15b5)

Reply:
- Well resolved 
<img width="432" alt="Screenshot 2023-09-02 at 21 38 26" src="https://github.com/Innocent259/School-library/assets/58692590/0dc653ee-cd80-4f57-a549-5186469f2949">